docs/samples/browser-plugin-meetings/app.js

```diff
@@ -1003,31 +1005,146 @@ async function getUserMedia(constraints = {audio: true, video: true}) {
   if (!meeting) {
     console.log('MeetingControls#getUserMedia() :: no valid meeting object!');
 
-    return Promise.reject(new Error('No valid meeting object.'));
+    throw new Error('No valid meeting object.');
+  }
+
+  if (localMedia.microphoneStream) {
+    console.log('MeetingControls#getUserMedia() :: microphone stream is already available');
+  }
+  else if (constraints.audio) {
+    await loadMicrophone(constraints.audio);
+  }
+
+  if (localMedia.cameraStream) {
+    console.log('MeetingControls#getUserMedia() :: camera stream is already available');
+  }
+  else if (constraints.video) {
+    await loadCamera(constraints.video);
+  }
+
+  console.log('MeetingControls#getUserMedia() :: following local streams are ready to use:', localMedia.microphoneStream, localMedia.cameraStream);
+}
+
+async function loadCamera(constraints) {
+  try {
+    console.log('MeetingControls#loadCamera() :: using webrtc-core local camera stream');
+    const videoConstraints = {...localMedia.videoConstraints[localVideoQuality[localResolutionInp.value]], ...constraints};
+
+    console.log('MeetingControls#loadCamera() :: getting camera stream with constraints: ', videoConstraints);
+    localMedia.cameraStream = await webex.meetings.mediaHelpers.createCameraStream(videoConstraints);
+
+    meetingStreamsLocalVideo.srcObject = localMedia.cameraStream.outputStream;
+    toggleVbgBtn.innerText = "Enable VBG";
+    loadCameraBtn.disabled = true;
+    console.log('MeetingControls#loadCamera() :: Successfully got camera stream:', localMedia.cameraStream);
+  }
+  catch (e) {
+    console.log('MeetingControls#loadCamera() :: Error getting camera stream!');
+    throw e;
+  }
+}
+
+async function handleVbg() {
+  try {
+    let effect;
+    const modeVal = document.getElementById('mode-type').value;
+    if (toggleVbgBtn.innerText === "Enable VBG") {
```

Comment:
I see the code is better now, but it could still be improved further. The `handleEffectsButton()` is called from many places and in each one you have hardcoded values for whether the effect is enabled or not and the title. Wouldn't it be better to just pass the effect instance into it and the `handleEffectsButton` function would actually read the  `effect.isEnabled` and set the title accordingly?

Reply:
Updated the code as per suggestion.